src/state.py

```diff
@@ -254,17 +300,19 @@ def from_charm(cls, charm: ops.CharmBase) -> "State":
             CharmRelationDataInvalidError: if invalid relation data was received.
             CharmIllegalNumUnitsError: if more than 1 unit of Jenkins charm is deployed.
         """
-        time_range_str = charm.config.get("restart-time-range")
-        if time_range_str:
-            try:
+        try:
+            remoting_config = RemotingConfig.from_config(config=charm.config)
```

Comment:
Maybe this could be outside of the try/catch block

Reply:
I needed to add it to the try-catch block to catch the `ValidationError` below. I tried separating the remoting config parsing and the restart timerange config into separate try-catch blocks but that'll make the function too complex and flake is not happy about that :sweat_smile: